src/main/java/com/kapeta/spring/config/KapetaSecurityProviderConfig.java

```diff
@@ -26,11 +38,44 @@ public SecurityFilterChain kapetaFilterChain(HttpSecurity http) throws Exception
                         .requestMatchers(KapetaAuthenticationRestController.PATH_KAPETA_AUTHENTICATION).permitAll()
                         .anyRequest().authenticated()
                 );
+
+        return http.build();
+    }
+
+    @Bean
+    public SecurityFilterChain jwksFilterChain(HttpSecurity http) throws Exception {
+        http
+                .securityMatcher(KapetaJwksRestController.PATH_WELL_KNOWN_JWKS)
```

Comment:
This ensures that this particular security chain only applies to this specific path